File: unidbg-api/src/main/java/com/vortexdbg/debugger/gdb/StepCommand.java
package com.vortexdbg.debugger.gdb;

import com.vortexdbg.Emulator;

class StepCommand implements GdbStubCommand {

    @Override
    public boolean processCommand(Emulator<?> emulator, GdbStub stub, String command) {
        stub.singleStep();
        stub.makePacketAndSend("OK");
        return true;
    }

}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/debugger/gdb/StepCommand.kt
package com.vortexdbg.debugger.gdb

import com.vortexdbg.Emulator

internal class StepCommand : GdbStubCommand {

    override fun processCommand(emulator: Emulator<*>, stub: GdbStub, command: String): Boolean {
        stub.singleStep()
        stub.makePacketAndSend("OK")
        return true
    }

}
